### PR TITLE
ENCD-6036 reset chr range in cart

### PR DESCRIPTION
**Pull request by emro**
"sortAndRefresh" is getting called multiple times when the genome assembly/annotation are updated - first for the assembly/annotation state update and then for the files state update - however the first call doesn't finish executing before the second commences, causing incorrect files to be displayed. Deferring "sortAndRefresh" till the call stack is clear resolves this.

### CODE REVIEW

File: src/encoded/static/components/genome_browser.js [hunk 1]
```diff
@@ -703,23 +703,28 @@ class GenomeBrowser extends React.Component {
     /* eslint-disable react/no-did-update-set-state */
     componentDidUpdate(prevProps, prevState) {
         if (!(this.state.disableBrowserForIE) && this.GV) {
-            if (this.state.contig !== prevState.contig) {
+            const contigUpdate = this.state.contig !== prevState.contig;
```
Comment (zoldello) [inline src/encoded/static/components/genome_browser.js:706]: `contigUpdate` follows react-naming style of booleans so it is fine. But personally I prefer `IsContigUpdated`. WIth that, I know it is a boolean and would not be expected a value like a number or string.

File: src/encoded/static/components/genome_browser.js [hunk 2]
```diff
@@ -703,23 +703,28 @@ class GenomeBrowser extends React.Component {
...
+            const assemblyUpdate = this.props.assembly !== prevProps.assembly || this.props.annotation !== prevProps.annotation;
+            const filesUpdate = !(_.isEqual(this.props.files, prevProps.files));
```
Comment (zoldello) [inline src/encoded/static/components/genome_browser.js:708]: _.isEqual does a deep copy. Is that needed as a shallow copy is faster if it is possible.

**Review by zoldello — APPROVED** [review]
Code looks good. My comments are insignificant.

File: src/encoded/static/components/genome_browser.js (hunk 2 above)
Comment (emro) [reply src/encoded/static/components/genome_browser.js:708]: I think so, though I wrote this code a while ago.

**Pull request merged**